Week1.java

```diff
@@ -0,0 +1,54 @@
+package Homework;
+
+import java.util.HashMap;
+import java.util.Scanner;
+import java.lang.*;
+
+public class Week1 {
+	// validating the string for null and spaces. 
+	public static boolean ValidateString(String s) {
+		if(s==null) {
+			return false;
+		}
+		for(int i=0;i<s.length();i++) {

```

Comment:
This involves going through the string an extra time - you are already going through the string in the LongestLengthSubstring method, its more efficient to do it just once!